pkg/cli/tail.go

```diff
@@ -0,0 +1,254 @@
+type logPuller interface {
+	Pull(context.Context, string, int) ([]*loggingpb.LogEntry, error)
+}
+
+var _ cli.Command = (*TailCommand)(nil)
+
+// TailCommand tails and validates(optional) lumberjack logs.
+type TailCommand struct {
+	cli.BaseCommand
+
+	flagResource string
+
+	flagValidate bool
+
+	flagMaxNum int
+
+	flagDuration time.Duration
+
+	flagCustomQuery string
+
+	flagRemoveLumberjackLogType bool
+
+	flagValidateWithAdditionalCheck bool
+
+	// For testing only.
+	testPuller logPuller
+}
+
+func (c *TailCommand) Desc() string {
+	return `Tail lumberjack logs from GCP Cloud logging`
+}
+
+func (c *TailCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+Tails and validates the latest lumberjack log in the last 24 hours from resource:
+
+      {{ COMMAND }} -resource "project/foo" -validate
+
+Tails the latest lumberjack log filtered by additional custom query:
+
+      {{ COMMAND }} -resource "project/foo" -query "resource.type = \"foo\""
+
+Tails and validates (with additional check) the latest 10 lumberjack log in the last 2 hours from resource:
+
+      {{ COMMAND }} -resource "project/foo" -max-num 10 -duration 2h -validate-with-additional-check
+
+Pulls and validates the latest non-lumberjack log type log:
+
+      {{ COMMAND }} -resource "project/foo" -remove-lumberjack-log-type
+`
+}
+
+func (c *TailCommand) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet()
+
+	// Command options
+	f := set.NewSection("COMMAND OPTIONS")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "resource",
+		Aliases: []string{"r"},
+		Target:  &c.flagResource,
+		Example: `projects/foo`,
+		Usage: `Name of the parent resource from which to retrieve log entries,` +
+			`examples are: projects/[PROJECT_ID], folders/[FOLDER_ID],` +
+			`organizations/[ORGANIZATION_ID], billingAccounts/[BILLING_ACCOUNT_ID]`,
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "validate",
+		Aliases: []string{"v"},
+		Target:  &c.flagValidate,
+		Default: false,
+		Usage:   `Turn on for lumberjack log validation`,
+	})
+
+	f.IntVar(&cli.IntVar{
+		Name:    "max-num",
+		Aliases: []string{"n"},
+		Target:  &c.flagMaxNum,
+		Default: 1,
+		Usage:   `Maximum number of most recent logs to validate, default is 1`,
+	})
+
+	f.DurationVar(&cli.DurationVar{
+		Name:    "duration",
+		Aliases: []string{"d"},
+		Target:  &c.flagDuration,
+		Example: "2h",
+		Default: 24 * time.Hour,
+		Usage:   `How far back to search for log entries, default is 24 hours`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "query",
+		Target:  &c.flagCustomQuery,
+		Example: `resource.type = "gae_app" AND severity = ERROR`,
+		Usage: `Optional custom log queries, see more on ` +
+			`https://cloud.google.com/logging/docs/view/logging-query-language`,
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "remove-lumberjack-log-type",
+		Target:  &c.flagRemoveLumberjackLogType,
+		Default: false,
+		Usage:   `Turn on to remove lumberjack log type log filter`,
+	})
```

Comment:
I found it useful when I am testing the cli and I don't have lumberjack logs in order to see the output. It might also be helpful when users forget to add the log type and want to query specific test logs?

Reply:
update to use override-filter flag, let me know what you think!

Reply:
Resolving, feel free to reopen